native/components/message-markdown-rules.react.js

```diff
@@ -1,110 +1,102 @@
 // @flow
 
+import type {
+  State,
+  SingleASTNode,
+  Output,
+  ReactElement,
+} from '../types/markdown';
+import type { StyleSheetOf } from '../themes/colors';
+import type { Style } from '../themes/markdown-styles';
+
 import * as React from 'react';
 import { Text } from 'react-native';
 import SimpleMarkdown from 'simple-markdown';
 
-type State = { [string]: any };
-
-type ASTNode = SingleASTNode | Array<SingleASTNode>;
-
-type SingleASTNode = {
-  type: string,
-  [string]: any,
-};
-
-type ReactElement = React$Element<any>;
-
-type Output<Result> = (node: ASTNode, state?: ?State) => Result;
-
-export default function(styles: any, opts: any = {}) {
+export default function rules(
+  styles: StyleSheetOf<Style>,
+  onLink: (url: string) => Promise<any>,
+) {
   const LINK_INSIDE = '(?:\\[[^\\]]*\\]|[^\\]]|\\](?=[^\\[]*\\]))*';
   const LINK_HREF_AND_TITLE =
     '\\s*<?([^\\s]*?)>?(?:\\s+[\'"]([\\s\\S]*?)[\'"])?\\s*';
-  var pressHandler = function(target) {
-    if (opts.onLink) {
-      opts.onLink(target).catch(function(error) {
+  const pressHandler = target => {
+    if (onLink) {
+      onLink(target).catch(error => {
         console.log(
           'There has been a problem with this action. ' + error.message,
         );
         throw error;
       });
     }
   };
+
   return {
     autolink: {
-      react: function(
+      ...SimpleMarkdown.defaultRules.autolink,
+      react: (
         node: SingleASTNode,
         output: Output<ReactElement>,
         state: State,
-      ) {
+      ) => {
         state.withinText = true;
         const _pressHandler = () => {
           pressHandler(node.target);
         };
-        const autolink = React.createElement(
-          Text,
-          {
-            key: state.key,
-            style: styles.autolink,
-            onPress: _pressHandler,
-          },
-          output(node.content, state),
+        const autolink = (
+          <Text key={state.key} style={styles.autolink} onPress={_pressHandler}>
+            {output(node.content, state)}
+          </Text>
         );
         return autolink;
       },
     },
     link: {
+      ...SimpleMarkdown.defaultRules.link,
       match: SimpleMarkdown.inlineRegex(
         new RegExp(
           '^\\[(' + LINK_INSIDE + ')\\]\\(' + LINK_HREF_AND_TITLE + '\\)',
         ),
       ),
-      react: function(
-        node: SingleASTNode,
-        output: Output<ReactElement>,
-        state: State,
-      ) {
+      react: (
+        node: SimpleMarkdown.SingleASTNode,
+        output: SimpleMarkdown.Output<ReactElement>,
```

Comment:
I think importing like this is making this `any`, so let me know which option would be better to use

Reply:
I'm confused - importing what is making what `any`?

Reply:
When I don't specify those types, but import from `simple-markdown`, flow is treating them as `any`

Reply:
Here's how I would've asked this question:

> I am not sure how to import the types from `simple-markdown`. I've noticed they're defined in [this file](https://github.com/Khan/simple-markdown/blob/master/simple-markdown.js), but when I try using them like this:
> ```js
> import SimpleMarkdown from 'simple-markdown';
> let output: SimpleMarkdown.Output<ReactElement>...
> ```
> I see that `output` is getting typed as `any`. Am I doing something wrong here?

Here are some things you could try:
1. `import * as SimpleMarkdown from 'simple-markdown';`
2. `import type { Output } from 'simple-markdown/simple-markdown';`